Added the exception object to the ``TestReport``, for reference in hook functions.